projects/common/src/telemetry/user-telemetry-internal.service.ts

```diff
@@ -0,0 +1,155 @@
+      this.telemetryProviders = [...this.telemetryProviders, ...providers];
+    } catch (error) {
+      /**
+       * NoOp
+       */
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.destroyedSubject.next();
+    this.destroyedSubject.complete();
+  }
+
+  public identify(userTraits: UserTraits): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.Identify, data: userTraits });
+  }
+
+  public shutdown(): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.Shutdown });
+  }
+
+  public trackEvent(name: string, data: Dictionary<unknown>): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.TrackEvent, name: name, data: data });
+  }
+
+  public trackPageEvent(url: string, data: Dictionary<unknown>): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.TrackPageView, name: url, data: data });
+  }
+
+  public trackErrorEvent(error: string, data: Dictionary<unknown>): void {
+    this.telemetryActionSubject.next({ type: TelemetryActionType.TrackError, name: error, data: data });
+  }
+
+  private buildTelemetryProvider(config: UserTelemetryRegistrationConfig<unknown>): UserTelemetryInternalConfig {
+    const providerInstance = this.injector.get(config.telemetryProvider);
+    providerInstance.initialize(config.initConfig);
+
+    return {
+      ...config,
+      telemetryProvider: providerInstance
+    };
+  }
+
+  private setupTelemetryActionHandlers(): void {
+    this.setupIdentifyHandler();
+    this.setupTrackEventHandler();
+    this.setupTrackPageHandler();
+    this.setupTrackErrorHandler();
+  }
+
+  private setupIdentifyHandler(): void {
+    this.identifyAction$.subscribe(action =>
+      this.telemetryProviders.forEach(provider => provider.telemetryProvider.identify(action.data as UserTraits))
+    );
+  }
+
+  private setupTrackEventHandler(): void {
+    this.trackEventAction$.subscribe(action =>
+      this.telemetryProviders
+        .filter(provider => provider.enableEventTracking)
+        .forEach(provider => provider.telemetryProvider.trackEvent?.(action.name!, action.data!))
+    );
+  }
+
+  private setupTrackPageHandler(): void {
+    this.trackPageAction$.subscribe(action =>
+      this.telemetryProviders
+        .filter(provider => provider.enablePageTracking)
+        .forEach(provider => provider.telemetryProvider.trackPage?.(action.name!, action.data!))
+    );
+  }
+
+  private setupTrackErrorHandler(): void {
+    this.trackErrorAction$.subscribe(action =>
+      this.telemetryProviders
+        .filter(provider => provider.enableErrorTracking)
+        .forEach(provider => provider.telemetryProvider.trackError?.(`Error: ${action.name!}`, action.data!))
+    );
+  }
+
+  private setupAutomaticPageTracking(): void {
+    this.router.events
+      .pipe(
+        filter((event): event is NavigationEnd => event instanceof NavigationEnd),
+        map(route => this.trackPageEvent(`Visited: ${route.url}`, { url: route.url })),
```

Comment:
this should be in the subscription rather than mapping for side effects.

Reply:
yes, I missed to update this one